android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java

```diff
@@ -321,6 +314,23 @@ private void deliverEventToGestureHandler(GestureHandler handler, MotionEvent ev
     }
   }
 
+  /**
+   * isViewAttachedUnderWrapper checks whether all of parents for view related to handler
+   * view are attached. Since there might be an issue rarely observed when view
+   * has been detached and handler's state hasn't been change to canceled, failed or
+   * ended yet. Probably it's a result of some race condition and stopping delivering
+   * for this handler and changing its state to failed of end appear to be good enough solution.
+   */
+  private boolean isViewAttachedUnderWrapper(@Nullable View view) {
+    while (view != mWrapperView) {
```

Comment:
`getParent` method is a part of `ViewParent` interface, therefore we don't need casting or instanceof checks:

```
    @Nullable ViewParent parent = view;
    while (parent != null && parent != mWrapperView) {
        parent = parent.getParent();
    }
    return parent == mWrapperView;
```

Reply:
Ok, I'll do it, but what was do reason you've done it like this before?

Reply:
And can't `view` do not fill implement `ViewParent` interface?